oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/pipelined/NodeStateEntryBatchTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.index.indexer.document.flatfile.pipelined;
+
+import org.junit.Test;
+
+import java.nio.BufferOverflowException;
+import java.nio.ByteBuffer;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class NodeStateEntryBatchTest {
+
+    @Test
+    public void testMaximumNumberOfEntries() {
+        NodeStateEntryBatch batch = NodeStateEntryBatch.createNodeStateEntryBatch(1024, 2);
+        assertFalse(batch.isAtMaxEntries());
+        batch.addEntry("a", new byte[1]);
+        assertFalse(batch.isAtMaxEntries());
+        batch.addEntry("b", new byte[1]);
+        assertEquals(2, batch.numberOfEntries());
+        assertTrue(batch.isAtMaxEntries());
+        try {
+            batch.addEntry("c", new byte[1]);
+            throw new AssertionError("Expected exception");
+        } catch (IllegalStateException e) {
+            // expected
+        }
+    }
```

Comment:
what about using `expected`? Same for the test below.
```suggestion
    @Test(expected = IllegalStateException.class)
    public void testMaximumNumberOfEntries() {
        NodeStateEntryBatch batch = NodeStateEntryBatch.createNodeStateEntryBatch(1024, 2);
        assertFalse(batch.isAtMaxEntries());
        batch.addEntry("a", new byte[1]);
        assertFalse(batch.isAtMaxEntries());
        batch.addEntry("b", new byte[1]);
        assertEquals(2, batch.numberOfEntries());
        assertTrue(batch.isAtMaxEntries());
        batch.addEntry("c", new byte[1]);
    }```

Reply:
I replaced by something like:
```assertThrows(BufferOverflowException.class, () -> batch.addEntry("b", new byte[1]));```
which is supported since JUnit 4.13. Oak is using 4.13.1.